overwatch-rs/src/services/handle.rs

```diff
@@ -133,17 +153,28 @@ impl<S: ServiceData> ServiceHandle<S> {
     }
 }
 
-impl<S: ServiceData> ServiceStateHandle<S> {
+impl<Service> ServiceStateHandle<Service::Message, Service::Settings, Service, Service::State>
+where
+    Service: ServiceData,
+{
+    #[must_use]
     pub fn id(&self) -> ServiceId {
-        S::SERVICE_ID
+        Service::SERVICE_ID
     }
 }
 
-impl<S> ServiceRunner<S>
+impl<Service>
+    ServiceRunner<
+        Service::Message,
+        Service::Settings,
+        Service,
+        Service::State,
+        Service::StateOperator,
+    >
 where
-    S::State: Send + Sync + 'static,
-    S::StateOperator: Send + 'static,
-    S: ServiceCore + 'static,
+    Service: ServiceCore + 'static,
+    Service::State: Clone + Send + Sync + 'static,
```

Comment:
Actually the whole point was to remove unnecessary bound. We are just moving them one upper layer which I think is good. But can we get rid of the `Clone` here?

Reply:
Ok, in this case, comes up with our decision. If we want a single responsible for the state, there should not be clonable. But, when implementing this in earlier stages the idea was for anyone to be able to actually get an update if needed. :thinking: 
Wdyt? We could implement our own non-clonable watcher.

Reply:
Yes we can explore that. Could that be tackled in a different PR? Do you want me to make a ticket for it?

Reply:
Yeah, for another PR. Go ahead with the ticket. Thanks!